deploy/stacks/lambda_api.py

```diff
@@ -425,15 +493,38 @@ def set_up_graphql_api_gateway(
         apig_vpce,
         resource_prefix,
         user_pool,
+        custom_auth
     ):
-        cognito_authorizer = apigw.CognitoUserPoolsAuthorizer(
-            self,
-            'CognitoAuthorizer',
-            cognito_user_pools=[user_pool],
-            authorizer_name=f'{resource_prefix}-{envname}-cognito-authorizer',
-            identity_source='method.request.header.Authorization',
-            results_cache_ttl=Duration.minutes(60),
-        )
+        if custom_auth is None:
+            cognito_authorizer = apigw.CognitoUserPoolsAuthorizer(
+                self,
+                'CognitoAuthorizer',
+                cognito_user_pools=[user_pool],
+                authorizer_name=f'{resource_prefix}-{envname}-cognito-authorizer',
+                identity_source='method.request.header.Authorization',
+                results_cache_ttl=Duration.minutes(60),
+            )
+        else:
+            #Create a custom Authorizer
+            custom_authorizer_role = iam.Role(self,
+                                              'custom-auth-role',
```

Comment:
Can add the `role_name` parameter in CDK and set it to `{resource_prefix}-{envname}-custom....`? If we want to restrict CDK policies in the future having prefixed resources with controlled names will make our life easier

Reply:
Updated code with the `role_name` changes